### PR TITLE
Final changes before first 2.1.0 release candidate

### PR DESCRIPTION
**Pull request by kgaillot**
This includes issues found by static analysis, routine release preparation, and various small fixes I wanted to squeeze in the release.

### CODE REVIEW

**Comment by kgaillot** [comment]
@clumens , can you review this when you get a chance? Thanks

**Comment by clumens** [comment]
You'll want to remove travisci_build_coverity_scan.sh from maint/Makefile.am.

File: lib/lrmd/lrmd_client.c [hunk 1]
```diff
@@ -1672,7 +1673,10 @@ lrmd_api_get_recurring_ops(lrmd_t *lrmd, const char *rsc_id, int timeout_ms,
 
             lrmd_op_info_t *op_info = calloc(1, sizeof(lrmd_op_info_t));
 
-            CRM_CHECK(op_info != NULL, break);
+            if (op_info == NULL) {
+                rc = -ENOMEM;
+                break;
```
Comment (clumens) [inline lib/lrmd/lrmd_client.c:1678]: This is only going to break out of the inner for loop.  Perhaps this is meant to be a return instead?

File: lib/cluster/cpg.c [hunk 2]
```diff
@@ -387,13 +387,14 @@ check_message_sanity(const pcmk__cpg_msg_t *msg)
...
-        if (msg->size != str_size) {
-            crm_warn("Message payload is corrupted: expected %llu bytes, got %llu",
-                     (unsigned long long) msg->size,
-                     (unsigned long long) str_size);
+    if (sane && !msg->is_compressed) {
+        /* msg->size == strlen(msg->data) + 1 would be a stronger check,
+         * but checking the last byte or two should be quick
+         */
+        if (((msg->size > 1) && (msg->data[msg->size - 2] == '\0'))
+            || (msg->data[msg->size - 1] != '\0')) {
```
Comment (clumens) [inline lib/cluster/cpg.c]: Is it possible for msg->size to be 0?

File: lib/cluster/cpg.c [hunk 3]
```diff
@@ -362,58 +362,66 @@ msg_type2text(enum crm_ais_msg_types type)
...
-                msg->sender.pid, msg->is_compressed, msg_data_len(msg), msg->header.size);
-
-    } else {
-        crm_trace
-            ("Verified message %d: (dest=%s:%s, from=%s:%s.%u, compressed=%d, size=%d, total=%d)",
-             msg->id, ais_dest(&(msg->host)), msg_type2text(dest), ais_dest(&(msg->sender)),
-             msg_type2text(msg->sender.type), msg->sender.pid, msg->is_compressed,
-             msg_data_len(msg), msg->header.size);
+        (((msg->size > 1) && (msg->data[msg->size - 2] == '\0'))
+         || (msg->data[msg->size - 1] != '\0'))) {
```
Comment (clumens) [inline lib/cluster/cpg.c:408]: Same - can msg->size be 0?

File: lib/lrmd/lrmd_client.c (hunk 1 above)
Comment (kgaillot) [reply lib/lrmd/lrmd_client.c:1678]: Good point, I think the idea was to keep trying later entries, but if we can't allocate a tiny amount of memory there's no point

File: lib/cluster/cpg.c (hunk 2 above)
Comment (kgaillot) [reply lib/cluster/cpg.c]: We bail earlier for that by the combination of (payload_size < 1) and (msg_data_len(msg) != payload_size)

**Comment by kgaillot** [comment]
Updated "travis helper" and "correctly check" commits

**Comment by kgaillot** [comment]
Rebased on current master and added one commit on top after testing

**Comment by clumens** [comment]
This all looks good to me.

**Pull request merged**